tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py

```diff
@@ -59,6 +59,17 @@ def test_lu_factor_reconstruction(self, dtype):
         cupy.testing.assert_allclose(LU, PA, atol=1e-5)
 
 
+@testing.gpu
+@testing.with_requires('scipy')
+class TestLUFactorSingular:
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(scipy_name='scp')
```

Comment:
`numpy_cupy_allclose` is OK just for the matrix `[[0, 1], [0, 0]]` (rank=n-1 and no numerical error).

```
a = rng.normal(size=(100, 50)) @ rng.normal(size=(50, 100))  # 100x100, rank-50
>>> scipy.linalg.lu_factor(a)[1]
array([26, 34,  9, 37, 29, 73, 35, 99, 14, 46, 15, 13, 89, 56, 70, 46, 98,
       63, 88, 84, 71, 21, 76, 68, 41, 57, 29, 39, 80, 39, 60, 66, 60, 49,
       78, 69, 65, 72, 50, 56, 78, 81, 90, 73, 92, 94, 85, 82, 77, 62, 87,
       93, 65, 97, 57, 86, 95, 71, 96, 79, 61, 86, 67, 73, 75, 74, 85, 79,
       85, 97, 73, 93, 89, 82, 82, 82, 89, 86, 84, 94, 89, 98, 86, 94, 99,
       90, 95, 95, 96, 98, 93, 95, 92, 98, 98, 99, 96, 99, 98, 99],
      dtype=int32)
>>> cupyx.scipy.linalg.lu_factor(cupy.array(a))[1]
array([26, 34,  9, 37, 29, 73, 35, 99, 14, 46, 15, 13, 89, 56, 70, 46, 98,
       63, 88, 84, 71, 21, 76, 68, 41, 57, 29, 39, 80, 39, 60, 66, 60, 49,
       78, 69, 65, 72, 50, 56, 78, 81, 90, 73, 92, 94, 85, 82, 77, 62, 64,
       97, 59, 71, 94, 67, 94, 61, 96, 82, 89, 91, 71, 96, 65, 85, 75, 90,
       77, 95, 76, 84, 90, 76, 91, 99, 82, 78, 87, 85, 81, 86, 93, 97, 93,
       96, 98, 96, 96, 95, 93, 94, 97, 94, 99, 98, 99, 99, 99, 99],
      dtype=int32)
```

Comment why the test is OK or test only the reconstructed matrix like `test_lu_factor_reconstruction` above.

To generate a singular matrix, use `cupy.testing.helper.generate_matrix` or follow https://github.com/cupy/cupy/blob/ba8c3fff19b322d5392311312d944b66c84cddf5/tests/cupy_tests/linalg_tests/test_solve.py#L194-L197

Reply:
Thanks! I'll fix the test as you show the right way.

Reply:
PTAL, @toslunar! BTW, can I have why the operation above brings a matrix singular?

Reply:
Multiplying the `ones` vector (from left or from right) results zero.  Therefore, `matrix_rank(a) <= min(m-1, n-1) < min(m, n)`.